zokrates_cli/src/ops/mpc/beacon.rs

```diff
@@ -0,0 +1,152 @@
+use crate::constants::MPC_DEFAULT_PATH;
+use clap::{App, Arg, ArgMatches, SubCommand};
+use phase2::parameters::MPCParameters;
+use std::fs::File;
+use std::io::{BufReader, BufWriter};
+use std::path::Path;
+
+pub fn subcommand() -> App<'static, 'static> {
+    SubCommand::with_name("beacon")
+        .about("Applies a random beacon")
+        .display_order(3)
+        .arg(
+            Arg::with_name("input")
+                .short("i")
+                .long("input")
+                .help("Path of the MPC parameters")
+                .value_name("FILE")
+                .takes_value(true)
+                .required(false)
+                .default_value(MPC_DEFAULT_PATH),
+        )
+        .arg(
+            Arg::with_name("hash")
+                .short("h")
+                .long("hash")
+                .help("Hash used for the beacon")
+                .takes_value(true)
+                .required(true),
+        )
+        .arg(
+            Arg::with_name("iterations")
+                .short("n")
+                .long("iterations")
+                .help("Number of hash iterations")
+                .takes_value(true)
+                .required(true),
+        )
+        .arg(
+            Arg::with_name("output")
+                .short("o")
+                .long("output")
+                .help("Path of the output file")
+                .value_name("FILE")
+                .takes_value(true)
+                .required(false)
+                .default_value(MPC_DEFAULT_PATH),
+        )
+}
+
+pub fn exec(sub_matches: &ArgMatches) -> Result<(), String> {
+    let path = Path::new(sub_matches.value_of("input").unwrap());
+    let file =
+        File::open(&path).map_err(|why| format!("Could not open `{}`: {}", path.display(), why))?;
+
+    let reader = BufReader::new(file);
+    let mut params = MPCParameters::read(reader, true)
+        .map_err(|why| format!("Could not read `{}`: {}", path.display(), why))?;
+
+    let beacon_hash = sub_matches.value_of("hash").unwrap();
+    let num_iterations: usize = sub_matches.value_of("iterations").unwrap().parse().unwrap();
+
+    if !(10..=63).contains(&num_iterations) {
+        return Err("Number of hash iterations should be in [10, 63] range".to_string());
+    }
+
+    println!("Creating a beacon RNG");
+
+    // Create an RNG based on the outcome of the random beacon
+    let mut rng = {
+        use byteorder::{BigEndian, ReadBytesExt};
+        use crypto::digest::Digest;
+        use crypto::sha2::Sha256;
+        use rand::chacha::ChaChaRng;
+        use rand::SeedableRng;
+
+        // The hash used for the beacon
+        let hash_result = hex::decode(beacon_hash);
```

Comment:
this would be more idiomatic with

```
let hash_result = hex::decode(beacon_hash)
   .map_err(|_| "Beacon hash should be in hexadecimal format".to_string())?;
```